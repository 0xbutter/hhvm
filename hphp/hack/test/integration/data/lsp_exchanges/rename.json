[
  {
    "jsonrpc": "2.0",
    "id": 1,
    "method": "initialize",
    "params": {
      "initializationOptions": {},
      "processId": null,
      "rootPath": "${root_path}",
      "capabilities": {}
    }
  },
  {
    "jsonrpc": "2.0",
    "method": "textDocument/didOpen",
    "params": {
      "textDocument": {
        "uri": "${php_file_uri}",
        "languageId": "hack",
        "version": 1,
        "text": "${php_file}"
      }
    }
  },
  {
    "jsonrpc": "2.0",
    "id": 2,
    "method": "textDocument/rename",
    "params": {
      "textDocument": {
        "uri": "${php_file_uri}"
      },
      "position": {
        "line": 9,
        "character": 42
      },
      "newName": "NEW_CONSTANT"
    }
  },
  {
    "jsonrpc": "2.0",
    "id": 3,
    "method": "textDocument/rename",
    "params": {
      "textDocument": {
        "uri": "${php_file_uri}"
      },
      "position": {
        "line": 2,
        "character": 11
      },
      "newName": "HistoryClass"
    }
  },
  {
    "jsonrpc": "2.0",
    "id": 4,
    "method": "textDocument/rename",
    "params": {
      "textDocument": {
        "uri": "${php_file_uri}"
      },
      "position": {
        "line": 19,
        "character": 24
      },
      "newName": "some_other_method"
    }
  },
  {
    "jsonrpc": "2.0",
    "id": 5,
    "method": "textDocument/rename",
    "params": {
      "textDocument": {
        "uri": "${php_file_uri}"
      },
      "position": {
        "line": 21,
        "character": 33
      },
      "newName": "ANOTHER_STR_CONSTANT"
    }
  },
  {
    "jsonrpc": "2.0",
    "id": 6,
    "method": "textDocument/rename",
    "params": {
      "textDocument": {
        "uri": "${php_file_uri}"
      },
      "position": {
        "line": 22,
        "character": 40
      },
      "newName": "AVOGADRO_CONSTANT"
    }
  },
  {
    "jsonrpc": "2.0",
    "id": 7,
    "method": "textDocument/rename",
    "params": {
      "textDocument": {
        "uri": "${php_file_uri}"
      },
      "position": {
        "line": 29,
        "character": 24
      },
      "newName": "$kittens_and_rainbows"
    }
  },
  {
    "jsonrpc": "2.0",
    "id": 999,
    "method": "shutdown",
    "params": {}
  }
]
