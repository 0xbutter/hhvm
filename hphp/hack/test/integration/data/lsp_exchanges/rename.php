<?hh  //strict
// comment
function a_rename(): int {
  return b_rename();
}

function b_rename(): int {
  return 42;
}

class TestClass {
  public function __construct(int $i) {}

  public function test_method(): int {
    return 1;
  }
}

function test_rename(): void {
  $test_class = new TestClass(1);
  $test_class->test_method(); $test_class->test_method();
}
