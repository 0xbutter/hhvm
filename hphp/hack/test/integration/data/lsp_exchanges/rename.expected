[
  {
    "jsonrpc": "2.0",
    "id": 1,
    "result": {
      "capabilities": {
        "textDocumentSync": {
          "openClose": true,
          "change": 2,
          "willSave": false,
          "willSaveWaitUntil": false,
          "save": {
            "includeText": false
          }
        },
        "hoverProvider": true,
        "completionProvider": {
          "resolveProvider": true,
          "triggerCharacters": [
            "$",
            ">",
            "\\",
            ":",
            "<"
          ]
        },
        "signatureHelpProvider": {
          "triggerCharacters": [
            "(",
            ","
          ]
        },
        "definitionProvider": true,
        "referencesProvider": true,
        "documentHighlightProvider": true,
        "documentSymbolProvider": true,
        "workspaceSymbolProvider": true,
        "codeActionProvider": false,
        "documentFormattingProvider": true,
        "documentRangeFormattingProvider": true,
        "documentOnTypeFormattingProvider": {
          "firstTriggerCharacter": ";",
          "moreTriggerCharacter": [
            "}"
          ]
        },
        "renameProvider": true,
        "typeCoverageProvider": true,
        "rageProvider": true
      }
    }
  },
  {
    "jsonrpc": "2.0",
    "id": 2,
    "result": {
      "changes": {
        "${php_file_uri}": [
          {
            "range": {
              "start": {
                "line": 3,
                "character": 8
              },
              "end": {
                "line": 3,
                "character": 16
              }
            },
            "newText": "NEW_CONSTANT"
          },
          {
            "range": {
              "start": {
                "line": 20,
                "character": 22
              },
              "end": {
                "line": 20,
                "character": 30
              }
            },
            "newText": "NEW_CONSTANT"
          },
          {
            "range": {
              "start": {
                "line": 9,
                "character": 38
              },
              "end": {
                "line": 9,
                "character": 46
              }
            },
            "newText": "NEW_CONSTANT"
          }
        ]
      }
    }
  },
  {
    "jsonrpc": "2.0",
    "id": 3,
    "result": {
      "changes": {
        "${php_file_uri}": [
          {
            "range": {
              "start": {
                "line": 2,
                "character": 6
              },
              "end": {
                "line": 2,
                "character": 15
              }
            },
            "newText": "HistoryClass"
          },
          {
            "range": {
              "start": {
                "line": 22,
                "character": 19
              },
              "end": {
                "line": 22,
                "character": 28
              }
            },
            "newText": "HistoryClass"
          },
          {
            "range": {
              "start": {
                "line": 21,
                "character": 15
              },
              "end": {
                "line": 21,
                "character": 24
              }
            },
            "newText": "HistoryClass"
          },
          {
            "range": {
              "start": {
                "line": 20,
                "character": 11
              },
              "end": {
                "line": 20,
                "character": 20
              }
            },
            "newText": "HistoryClass"
          },
          {
            "range": {
              "start": {
                "line": 18,
                "character": 20
              },
              "end": {
                "line": 18,
                "character": 29
              }
            },
            "newText": "HistoryClass"
          },
          {
            "range": {
              "start": {
                "line": 9,
                "character": 27
              },
              "end": {
                "line": 9,
                "character": 36
              }
            },
            "newText": "HistoryClass"
          }
        ]
      }
    }
  },
  {
    "jsonrpc": "2.0",
    "id": 4,
    "result": {
      "changes": {
        "${php_file_uri}": [
          {
            "range": {
              "start": {
                "line": 12,
                "character": 18
              },
              "end": {
                "line": 12,
                "character": 29
              }
            },
            "newText": "some_other_method"
          },
          {
            "range": {
              "start": {
                "line": 19,
                "character": 43
              },
              "end": {
                "line": 19,
                "character": 54
              }
            },
            "newText": "some_other_method"
          },
          {
            "range": {
              "start": {
                "line": 19,
                "character": 15
              },
              "end": {
                "line": 19,
                "character": 26
              }
            },
            "newText": "some_other_method"
          }
        ]
      }
    }
  },
  {
    "jsonrpc": "2.0",
    "id": 5,
    "result": {
      "changes": {
        "${php_file_uri}": [
          {
            "range": {
              "start": {
                "line": 4,
                "character": 15
              },
              "end": {
                "line": 4,
                "character": 27
              }
            },
            "newText": "ANOTHER_STR_CONSTANT"
          },
          {
            "range": {
              "start": {
                "line": 21,
                "character": 26
              },
              "end": {
                "line": 21,
                "character": 38
              }
            },
            "newText": "ANOTHER_STR_CONSTANT"
          }
        ]
      }
    }
  },
  {
    "jsonrpc": "2.0",
    "id": 6,
    "result": {
      "changes": {
        "${php_file_uri}": [
          {
            "range": {
              "start": {
                "line": 5,
                "character": 17
              },
              "end": {
                "line": 5,
                "character": 33
              }
            },
            "newText": "AVOGADRO_CONSTANT"
          },
          {
            "range": {
              "start": {
                "line": 22,
                "character": 30
              },
              "end": {
                "line": 22,
                "character": 46
              }
            },
            "newText": "AVOGADRO_CONSTANT"
          }
        ]
      }
    }
  },
  {
    "jsonrpc": "2.0",
    "id": 7,
    "result": {
      "changes": {
        "${php_file_uri}": [
          {
            "range": {
              "start": {
                "line": 25,
                "character": 34
              },
              "end": {
                "line": 25,
                "character": 40
              }
            },
            "newText": "$kittens_and_rainbows"
          },
          {
            "range": {
              "start": {
                "line": 26,
                "character": 2
              },
              "end": {
                "line": 26,
                "character": 8
              }
            },
            "newText": "$kittens_and_rainbows"
          },
          {
            "range": {
              "start": {
                "line": 27,
                "character": 4
              },
              "end": {
                "line": 27,
                "character": 10
              }
            },
            "newText": "$kittens_and_rainbows"
          },
          {
            "range": {
              "start": {
                "line": 27,
                "character": 14
              },
              "end": {
                "line": 27,
                "character": 20
              }
            },
            "newText": "$kittens_and_rainbows"
          },
          {
            "range": {
              "start": {
                "line": 27,
                "character": 32
              },
              "end": {
                "line": 27,
                "character": 38
              }
            },
            "newText": "$kittens_and_rainbows"
          },
          {
            "range": {
              "start": {
                "line": 29,
                "character": 20
              },
              "end": {
                "line": 29,
                "character": 26
              }
            },
            "newText": "$kittens_and_rainbows"
          },
          {
            "range": {
              "start": {
                "line": 32,
                "character": 30
              },
              "end": {
                "line": 32,
                "character": 36
              }
            },
            "newText": "$kittens_and_rainbows"
          },
          {
            "range": {
              "start": {
                "line": 33,
                "character": 11
              },
              "end": {
                "line": 33,
                "character": 17
              }
            },
            "newText": "$kittens_and_rainbows"
          }
        ]
      }
    }
  },
  {
    "jsonrpc": "2.0",
    "id": 999,
    "result": null
  }
]
